packages/kit/types/config.d.ts

```diff
@@ -1,38 +1,99 @@
 import { UserConfig as ViteConfig } from 'vite';
 import { RecursiveRequired } from './helper';
-import { Logger, TrailingSlash } from './internal';
+import { HttpMethod, Logger, RouteSegment, TrailingSlash } from './internal';
 
-export interface AdapterUtils {
+export interface RouteDefinition {
+	type: 'page' | 'endpoint';
+	pattern: RegExp;
+	segments: RouteSegment[];
+	methods: HttpMethod[];
+}
+
+export interface AdapterEntry {
+	/**
+	 * A string that uniquely identifies a function and is used for deduplication.
```

Comment:
I was thinking JS function when I first read this, so maybe we should say serverless function here? But then I though possibly it's not even a serverless thing, so maybe call it an HTTP service?
```suggestion
	 * A string that uniquely identifies an HTTP service (e.g. serverless function) and is used for deduplication.
```

Reply:
yeah, i like that